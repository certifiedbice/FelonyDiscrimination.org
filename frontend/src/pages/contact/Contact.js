import React from 'react';
import ContactForm from '../../components/contactform/ContactForm'
import './Contact.css';

export default function Contact(){
	return(
			<section id="contact-form-container">
				<ContactForm/>
			</section>
	);
}
