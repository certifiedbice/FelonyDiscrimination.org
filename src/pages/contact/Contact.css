#contact-form-container {
    flex: 1;
    text-align: center;
}

#contact-form {}

#contact-form legend {
    width: 100%;
    margin-bottom: 5px;
    font-size: 18px;
    font-weight: 700;
}

.contact-form-element-container {
    padding: 5px;   
}

.contact-form-element-container label {
    display: block;
    margin: 0 0 5px 20px;
    text-align: left;
}

.contact-form-element {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

.contact-form-element-container input {
    border-radius: 50px;
    padding: 5px;
}

.contact-form-element-container textarea {
    border-radius: 25px;
    padding: 10px;
}

#contact-form-submit {
    width: 90%;
    background-color: #4378bc;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
    cursor: pointer;
}

#contact-form-submit:hover {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

/* Desktop CSS */
@media(min-width:991px){
    
    .contact-form-element-container label {
        margin: 0 0 10px 60px;
    }

    #contact-form-submit {
        width: 25%;
    }

    #contact-form-submit:hover {
        width: 25%;
    }

}