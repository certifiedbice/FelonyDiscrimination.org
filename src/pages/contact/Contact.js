import React from 'react';
import './Contact.css';

export default function Contact(){
	return(
			<section id="contact-form-container">
				<form id="contact-form" name="contact-form" aria-label="Contact form">
					<fieldset>
						<legend><h2>Contact Us</h2></legend>
						<div className="contact-form-element-container">
							<label>Name: </label>
							<input className="contact-form-element" name="contact-form-name" id="contact-form-name" type="text" required aria-labelledby="Contact form name" placeholder="Name"/>
						</div>
						<div className="contact-form-element-container">
							<label>Phone: </label>
							<input className="contact-form-element" name="contact-form-phone" id="contact-form-phone" type="tel" aria-labelledby="Contact form phone" placeholder="Phone number"/>
						</div>
						<div className="contact-form-element-container">
							<label>Email: </label>
							<input className="contact-form-element" name="contact-form-email" id="contact-form-email" type="email" required aria-labelledby="Contact form email" placeholder="your@email.com"/>
						</div>
						<div className="contact-form-element-container">
							<label>Message: </label>
							<textarea className="contact-form-element" name="contact-form-message" id="contact-form-message" rows="6" cols="30" required aria-labelledby="Contact form message"/>
						</div>
						<div className="contact-form-element-container">
							<input id="contact-form-submit" type="submit" required aria-label="Contact form submit" value="Submit"/>
						</div>
					</fieldset>
				</form>
			</section>
	);
}
