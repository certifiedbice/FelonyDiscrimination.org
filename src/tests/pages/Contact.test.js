import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Contact from '../../pages/contact/Contact';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<Contact/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<Contact/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});