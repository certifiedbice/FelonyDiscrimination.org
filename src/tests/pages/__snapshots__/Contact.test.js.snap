// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="contact-form-container"
>
  <form
    aria-label="Contact form"
    id="contact-form"
    name="contact-form"
  >
    <fieldset>
      <legend>
        <h2>
          Contact Us
        </h2>
      </legend>
      <div
        className="contact-form-element-container"
      >
        <label>
          Name: 
        </label>
        <input
          aria-labelledby="Contact form name"
          className="contact-form-element"
          id="contact-form-name"
          name="contact-form-name"
          placeholder="Name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="contact-form-element-container"
      >
        <label>
          Phone: 
        </label>
        <input
          aria-labelledby="Contact form phone"
          className="contact-form-element"
          id="contact-form-phone"
          name="contact-form-phone"
          placeholder="Phone number"
          type="tel"
        />
      </div>
      <div
        className="contact-form-element-container"
      >
        <label>
          Email: 
        </label>
        <input
          aria-labelledby="Contact form email"
          className="contact-form-element"
          id="contact-form-email"
          name="contact-form-email"
          placeholder="your@email.com"
          required={true}
          type="email"
        />
      </div>
      <div
        className="contact-form-element-container"
      >
        <label>
          Message: 
        </label>
        <textarea
          aria-labelledby="Contact form message"
          className="contact-form-element"
          cols="30"
          id="contact-form-message"
          name="contact-form-message"
          required={true}
          rows="6"
        />
      </div>
      <div
        className="contact-form-element-container"
      >
        <input
          aria-label="Contact form submit"
          id="contact-form-submit"
          required={true}
          type="submit"
          value="Submit"
        />
      </div>
    </fieldset>
  </form>
</section>
`;
